lib/logitech_receiver/hidpp20.py

```diff
@@ -357,53 +359,276 @@ def __len__(self):
+                        ret.append(tgt_task)
+
+        return ret
+
+    @property
+    def mapping_flags(self) -> List[str]:
+        if self._mapping_flags is None:
+            self._getCidReporting()
+        return special_keys.MAPPING_FLAG.flag_names(self._mapping_flags)
+
+    def set_diverted(self, value: bool):
+        """If set, the control is diverted temporarily and reports presses as HID++ events
+        until a HID++ configuration reset occurs."""
+        self._setCidReporting(divert=value)
+
+    def set_persistently_diverted(self, value: bool):
+        """If set, the control is diverted permanently and reports presses as HID++ events."""
+        self._setCidReporting(persist=value)
+
+    def set_rawXY_reporting(self, value: bool):
+        """If set, the mouse reports all its raw XY events while this control is pressed
+        as HID++ events. Gets cleared on a HID++ configuration reset."""
+        self._setCidReporting(rawXY=value)
+
+    def remap(self, to: _NamedInt):
+        """Remaps this control to another action."""
+        self._setCidReporting(remap=int(to))
+
+    def _getCidReporting(self):
+        try:
+            mapped_data = feature_request(
+                self._device,
+                FEATURE.REPROG_CONTROLS_V4,
+                0x20,
+                *tuple(_pack('!H', self._cid)),
+            )
+            if mapped_data:
+                cid, mapping_flags, mapped_to = _unpack('!HBH', mapped_data[:5])
+                if cid != self._cid and _log.isEnabledFor(_WARNING):
+                    _log.warn(
+                        f'REPROG_CONTROLS_V4 endpoint getCidReporting on device {self._device} replied ' +
+                        f'with a different control ID ({cid}) than requested ({self._cid}).'
+                    )
+                self._mapping_flags = mapping_flags
+                self._mapped_to = mapped_to if mapped_to != 0 else self._cid
+            else:
+                raise FeatureCallError('No reply from device.')
+        except Exception:
+            if _log.isEnabledFor(_ERROR):
+                _log.error(f'Exception in _getCidReporting on device {self._device}: ', exc_info=1)
+                # Clear flags and set mapping target to self as fallback
+            self._mapping_flags = 0
+            self._mapped_to = self._cid
+
+    def _setCidReporting(self, divert=None, persist=None, rawXY=None, remap=0):
+        """Sends a `setCidReporting` request with the given parameters to the control. Raises
+        an exception if the parameters are invalid.
+        """
+        if rawXY:
+            # We need diversion to report raw XY, so divert temporarily
+            # (since XY reporting is also temporary)
+            divert = True
+
+        if divert is not None and special_keys.KEY_FLAG.divertable not in self.flags:
+            raise FeatureNotSupported(f'Tried to divert non-divertable control {self.key} ' + f'on device {self._device}.')
```

Comment:
The string split should probably be undone here and in a few other places.   The Solaar style has been set up for long line lengths (about 120 characters, I think).

Reply:
The line limit is set for 127. I am not sure if yapf allows the line to be split here, but if it does, please do it.